From 4e06df419c90c2f9a69dda57216b32fd9810e663 Mon Sep 17 00:00:00 2001
From: Bai Ping <ping.bai@nxp.com>
Date: Sun, 21 Feb 2016 14:53:50 +0800
Subject: [PATCH 0952/5242] MLK-12430 ARM: imx: enable and bypass pll1_bypass
 clk before changing arm_podf

commit  347accd133bd3621eb5bebce678b627eb31bdc85 from
https://source.codeaurora.org/external/imx/linux-imx.git

Before changing the ARM_PODF, the pll1_bypass clock should be enabled and
bypassed to make sure the ARM_PODF can be changed.

Signed-off-by: Bai Ping <ping.bai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/busfreq-imx.c |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-imx/busfreq-imx.c b/arch/arm/mach-imx/busfreq-imx.c
index 29d6896..928da37 100644
--- a/arch/arm/mach-imx/busfreq-imx.c
+++ b/arch/arm/mach-imx/busfreq-imx.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2011-2015 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright (C) 2011-2016 Freescale Semiconductor, Inc. All Rights Reserved.
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -368,6 +368,12 @@ static void enter_lpm_imx6sl(void)
 			clk_set_parent(pll1_sw_clk, step_clk);
 
 			/*
+			 * Need to ensure that PLL1 is bypassed and enabled
+			 * before ARM-PODF is set.
+			 */
+			clk_set_parent(pll1_bypass_clk, pll1_bypass_src_clk);
+
+			/*
 			 * Ensure that the clock will be
 			 * at original speed.
 			 */
-- 
1.7.9.5

